src/main/java/org/opentripplanner/model/plan/ScheduledTransitLeg.java

```diff
@@ -384,7 +386,7 @@ public String toString() {
       .addObjOp("agencyId", getAgency(), AbstractTransitEntity::getId)
       .addObjOp("routeId", getRoute(), AbstractTransitEntity::getId)
       .addObjOp("tripId", getTrip(), AbstractTransitEntity::getId)
-      .addStr("headsign", getHeadsign())
+      .addStr("headsign", headsign)
```

Comment:
We should probaby add an `addI18NStr`

Reply:
This also would throw archtitecture violation error. Should this be addressed before this pull request can be merged?

Reply:
```suggestion
      .addObj("headsign", getHeadsign())
```

Reply:
It is actually not needed, `addObj` does the correct thing